src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinNamesAnnotationIntrospector.kt

```diff
@@ -29,34 +29,34 @@ import kotlin.reflect.jvm.kotlinFunction
 
 internal class KotlinNamesAnnotationIntrospector(val module: KotlinModule, val cache: ReflectionCache, val ignoredClassesForImplyingJsonCreator: Set<KClass<*>>) : NopAnnotationIntrospector() {
     // since 2.4
-    override fun findImplicitPropertyName(member: AnnotatedMember): String? = when (member) {
-        is AnnotatedMethod -> if (member.name.contains('-') && member.parameterCount == 0) {
-            when {
-                member.name.startsWith("get") -> member.name.substringAfter("get")
-                member.name.startsWith("is") -> member.name.substringAfter("is")
-                else -> null
-            }?.replaceFirstChar { it.lowercase(Locale.getDefault()) }?.substringBefore('-')
-        } else null
-        is AnnotatedParameter -> findKotlinParameterName(member)
-        else -> null
-    }
-
-    // since 2.11: support Kotlin's way of handling "isXxx" backed properties where
-    // logical property name needs to remain "isXxx" and not become "xxx" as with Java Beans
-    // (see https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html and
-    //  https://github.com/FasterXML/jackson-databind/issues/2527
-    //  for details)
-    override fun findRenameByField(config: MapperConfig<*>,
-                                   field: AnnotatedField,
-                                   implName: PropertyName): PropertyName? {
-        val origSimple = implName.simpleName
-        if (field.declaringClass.isKotlinClass() && origSimple.startsWith("is")) {
-            val mangledName: String? = BeanUtil.stdManglePropertyName(origSimple, 2)
-            if ((mangledName != null) && !mangledName.equals(origSimple)) {
-                return PropertyName.construct(mangledName)
-            }
+    override fun findImplicitPropertyName(member: AnnotatedMember): String? {
+        if (!member.declaringClass.isKotlinClass()) return null
+
+        val name = member.name
+
+        return when (member) {
+            is AnnotatedMethod -> if (member.parameterCount == 0) {
+                // The reason for truncating after `-` is to truncate the random suffix
+                // given after the value class accessor name.
+                when {
+                    name.startsWith("get") -> name.takeIf { it.contains("-") }?.let { _ ->
```

Comment:
Ok: I _think_ you should ONLY need to consider `isXxx()` case, given that `getXxx()` should work by default.

The intent of "implicit name" was originally to allow exposing names for constructor parameters, because names were not included in byte code, unlike with methods and fields. So if there is no addition "implicit" name, underlying method/field name is used. It seems better (at least conceptually) to only override cases where there is difference.

Otherwise, I think that use of implicit name route may be better than rename using `findRenameByField`... if it works as well I see no reason for other case.

However: use of `findRenameByField()` may make sense for a different case: case where property name starts with capital letter. Intent with this method is that instead of starting with getters, setters and fields all to unify names, instead start with field name. This tends to work better for Kotlin, Scala etc where developers specify property name as "plain", similar to it being field (and in fact being added as `Field`) and getter/setter name is created based on that.

Reply:
> Ok: I _think_ you should ONLY need to consider `isXxx()` case, given that `getXxx()` should work by default.

This branching only applies when the getter name contains `-`, so I believe the behavior is as you say.
The reason for this is as per the comments on lines 39 and 40.

> I think that use of implicit name route may be better than rename using `findRenameByField`.

I don't see how using `findRenameByField` would avoid the problem (#340) of confusion when two properties, `isFoo` and `foo`, are defined (this is the main problem I wanted to solve).

Reply:
Ah. Ok, that `-` part I somehow missed although explicitly stated. Still, what I was trying to say is that I think you are right in preferring implicit name production over `findRenameByFiedl()` for cases you are thinking of solving.